app/controllers/IndexController.scala

```diff
@@ -38,21 +38,26 @@ class IndexController @Inject()(appConfig: FrontendAppConfig,
 
   def onPageLoad: Action[AnyContent] = identify.async {
     implicit request =>
-      destinationConnector.getArrivals() flatMap {
-        arrivals =>
-          val viewArrivals =
-            controllers.routes.ViewArrivalsController.onPageLoad().url
+      destinationConnector
+        .getArrivals()
+        .flatMap {
+          arrivals =>
+            val viewArrivals = controllers.routes.ViewArrivalsController.onPageLoad().url
 
-          renderer
-            .render(
-              "index.njk",
-              Json.obj(
-                "declareArrivalNotificationUrl" -> appConfig.declareArrivalNotificationUrl,
-                "viewArrivalNotificationUrl"    -> viewArrivals,
-                "hasArrivals"                   -> arrivals.arrivals.nonEmpty
+            renderer
+              .render(
+                "index.njk",
+                Json.obj(
+                  "declareArrivalNotificationUrl" -> appConfig.declareArrivalNotificationUrl,
+                  "viewArrivalNotificationUrl"    -> viewArrivals,
+                  "hasArrivals"                   -> arrivals.arrivals.nonEmpty
+                )
               )
-            )
-            .map(Ok(_))
-      }
+              .map(Ok(_))
+        }
+        .recover {
```

Comment:
@MatthewAlexanderBeck it is better to use recoverWith compare to recover, think about it.

Reply:
Hey Ashwini, my understanding was that recoverWith returns a future meaning this would need to be wrapped with a future.successful. Unless I am missing something?